admin/preprints/views.py

```diff
@@ -88,6 +88,120 @@ def get_context_data(self, **kwargs):
+        return PreprintRequest.objects.filter(
+            request_type='withdrawal',
+            target__guids___id=self.kwargs.get('guid'),
+            target__provider__reviews_workflow=None).first()
+
+class PreprintWithdrawalRequestList(PermissionRequiredMixin, ListView):
+
+    paginate_by = 10
+    paginate_orphans = 1
+    template_name = 'preprints/withdrawal_requests.html'
+    ordering = '-created'
+    permission_required = 'osf.change_preprintrequest'
+    raise_exception = True
+    context_object_name = 'preprintrequest'
+
+    def get_queryset(self):
+        return PreprintRequest.objects.filter(
+            request_type='withdrawal',
+            target__provider__reviews_workflow=None).exclude(
+                machine_state='initial').order_by(self.ordering)
+
+    def get_context_data(self, **kwargs):
+        query_set = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(query_set)
+        paginator, page, query_set, is_paginated = self.paginate_queryset(
+            query_set, page_size)
+        return {
+            'requests': map(serialize_withdrawal_request, query_set),
+            'page': page,
+        }
+
+    def post(self, request, *args, **kwargs):
+        if not request.user.has_perm('osf.change_preprintrequest'):
+            raise PermissionDenied('You do not have permission to approve or reject withdrawal requests.')
+        is_approve_action = 'approveRequest' in request.POST.keys()
+        request_ids = [
+            id_ for id_ in request.POST.keys()
+            if id_ not in ['csrfmiddlewaretoken', 'approveRequest', 'rejectRequest']
+        ]
+        for id_ in request_ids:
+            withdrawal_request = PreprintRequest.load(id_)
+            if is_approve_action:
+                withdrawal_request.run_accept(self.request.user, 'Approved by {}'.format(self.request.user.fullname))
+            else:
+                withdrawal_request.run_reject(self.request.user, 'Rejected by {}'.format(self.request.user.fullname))
+            update_admin_log(
+                user_id=self.request.user.id,
+                object_id=id_,
+                object_repr='PreprintRequest',
+                message='{} withdrawal request: {} of preprint {}'.format('Approved' if is_approve_action else 'Rejected', id_, withdrawal_request.target._id),
+                action_flag=APPROVE_WITHDRAWAL if is_approve_action else REJECT_WITHDRAWAL
+            )
+        return redirect('preprints:withdrawal-requests')
+
+
+class PreprintApproveWithdrawalRequest(PermissionRequiredMixin, PreprintRequestDeleteBase):
+    template_name = 'preprints/approve_withdrawal.html'
+    permission_required = 'osf.change_preprintrequest'
+    raise_exception = True
+
+    def post(self, request, *args, **kwargs):
+        withdrawal_request = self.get_object()
+        withdrawal_request.run_accept(self.request.user, 'Approved by {}'.format(self.request.user.fullname))
```

Comment:
Does this end up setting the "`withdrawal_justification`" to "Approved by [whoever]"?

Reply:
Is the request comment being set to the withdrawal justification at request creation? If so it would be overwritten by "Approved by [admin]". In reviews request, the moderator/admin adds the comment (justification of approval/rejection) that is what I had in mind.